src/power.rs

```diff
@@ -28,6 +40,76 @@ impl Power {
         Self { rb: pwr }
     }
 
+    pub fn get_standby_flag(&mut self) -> bool {
+        self.rb.sr1.read().sbf().bit_is_set()
+    }
+
+    pub fn get_wakeup_flag<L: Into<WakeUpLine>>(&self, lane: L) -> bool {
+        match lane.into() {
+            WakeUpLine::WKUP1 => self.rb.sr1.read().wuf1().bit_is_set(),
+            WakeUpLine::WKUP2 => self.rb.sr1.read().wuf2().bit_is_set(),
+            WakeUpLine::WKUP4 => self.rb.sr1.read().wuf4().bit_is_set(),
+            WakeUpLine::WKUP5 => self.rb.sr1.read().wuf5().bit_is_set(),
+            WakeUpLine::WKUP6 => self.rb.sr1.read().wuf6().bit_is_set(),
+            _ => false,
+        }
+    }
+
+    pub fn clear_wakeup_flag<L: Into<WakeUpLine>>(&mut self, lane: L) {
+        match lane.into() {
+            WakeUpLine::WKUP1 => self.rb.scr.write(|w| w.cwuf1().set_bit()),
+            WakeUpLine::WKUP2 => self.rb.scr.write(|w| w.cwuf2().set_bit()),
+            WakeUpLine::WKUP4 => self.rb.scr.write(|w| w.cwuf4().set_bit()),
+            WakeUpLine::WKUP5 => self.rb.scr.write(|w| w.cwuf5().set_bit()),
+            WakeUpLine::WKUP6 => self.rb.scr.write(|w| w.cwuf6().set_bit()),
+            _ => {}
+        }
+    }
+
+    pub fn clear_standby_flag(&mut self) {
+        self.rb.scr.write(|w| w.csbf().set_bit());
+    }
+
+    pub fn enable_wakeup_lane<L: Into<WakeUpLine>>(&mut self, lane: L, edge: SignalEdge) {
+        assert!(edge != SignalEdge::All);
+
+        let edge = edge == SignalEdge::Falling;
+        match lane.into() {
+            WakeUpLine::WKUP1 => {
+                self.rb.cr3.modify(|_, w| w.ewup1().set_bit());
+                self.rb.cr4.modify(|_, w| w.wp1().bit(edge));
+            }
+            WakeUpLine::WKUP2 => {
+                self.rb.cr3.modify(|_, w| w.ewup2().set_bit());
+                self.rb.cr4.modify(|_, w| w.wp2().bit(edge));
+            }
+            WakeUpLine::WKUP4 => {
+                self.rb.cr3.modify(|_, w| w.ewup4().set_bit());
+                self.rb.cr4.modify(|_, w| w.wp4().bit(edge));
+            }
+            WakeUpLine::WKUP5 => {
+                self.rb.cr3.modify(|_, w| w.ewup5().set_bit());
+                self.rb.cr4.modify(|_, w| w.wp4().bit(edge));
+            }
+            WakeUpLine::WKUP6 => {
+                self.rb.cr3.modify(|_, w| w.ewup6().set_bit());
+                self.rb.cr4.modify(|_, w| w.wp5().bit(edge));
```

Comment:
Did not check from datasheet but are `wp4` and `wp5` ok here (1, 2, 4 are different)?

Reply:
Oops, fixed